gpkit/nomials/math.py

```diff
@@ -371,6 +296,21 @@ class Monomial(Posynomial):
           These will be deprecated in the future, replaced with a single
           __init__ syntax, same as Signomial.
     """
+
+    @property
+    def exp(self):
+        "Creates exp or returns a cached exp"
+        if not self._exp:
+            self._exp, = self.hmap.keys()  # pylint: disable=attribute-defined-outside-init
+        return self._exp
```

Comment:
Yup; among other things there's a lot of Monomial-operations-then-check-equivalence tests.